serverless/containers/how-to/add-a-custom-domain-to-a-container.mdx

```diff
@@ -36,11 +36,65 @@ Each container comes with an automatically generated domain, on which it is avai
+
+## Migrate from a domain already serving another endpoint
+
+Read this section if you are trying to migrate a domain that is already serving a live website/API/web application -- but not yet hosted on a Serverless Container -- and you are trying to migrate.
+
+To clarify, let's take a concrete example:
+
+- your website is hosted on a Scaleway instance, with the IP `51.15.x.x`
+- you own a domain `mydomain.com`
+- an A record is configured on `website.mydomain.com` and points to `51.15.x.x`
+- when a client accesses `http://website.mydomain.com`, the request hits your instance IP `51.15.x.x:80`
+- you also have a running version of your website hosted as a Serverless Container (only accessible using `prodtd9zl1mz-website.functions.fnc.fr-par.scw.cloud`), and now you want your users to access this version from `http://website.mydomain.com`
+
+Before adding the custom domain on your Serverless Container, you must change the DNS record to point to the Serverless Container endpoint:
+
+- before: `website.mydomain.com` is an A record to `51.15.x.x`
+- after: `website.mydomain.com` is a CNAME record to `prodtd9zl1mz-website.functions.fnc.fr-par.scw.cloud`
+
+By doing this, clients that already have the `website.mydomain.com` DNS record cached locally will continue to hit `51.15.x.x`, until the TTL expires. New clients (or those whose cache has expired) will start to hit `prodtd9zl1mz-website.functions.fnc.fr-par.scw.cloud`. **However**, until the custom domain is not configured on the Serverless Container, these requests will end up in 404, because `website.mydomain.com` is not (yet) known in our infrastructure. Depending on your downtime tolerance (clients receiving 404) for a few minutes, there are 2 cases:
+
+- downtime is acceptable or can be planified (e.g. during the night or when there is less traffic on your website). In that case, after creating the CNAME record, and once you are sure DNS is propagated everywhere, create the custom domain in the Serverless Container. Once ready, requests to `http://website.mydomain.com` will hit your Serverless Container (no more 404).
+- downtime is not acceptable. Unfortunately, this is not possible right now out-of-the-box. To be able to answer to requests to `website.mydomain.com`, our infrastructure must know it, so custom domain has to be configured; but, for the custom domain to be configured, the DNS record must point to the Serverless Container endpoint, leaving us with a chicken-and-egg problem. To handle such scenarios, a third-party tool (like Cloudflare) can be configured to serve a cached version of your website while the domain is being reconfigured (for example with `stale-if-error` option). If you choose this solution, do not forget to disable cache on routes starting with `/.well-known/acme-challenge` to avoid issues described in the "Technical details and troubleshooting" section.
```

Comment:
nit: again just some wording, feel free to ignore

```suggestion
- downtime is not acceptable. Unfortunately, this is not possible right now out-of-the-box. To serve requests from `website.mydomain.com`, our infrastructure must know it, so a custom domain has to be configured. However, for the custom domain to be configured, the DNS record must point to the Serverless Container endpoint, resulting in a chicken-and-egg problem. To handle such scenarios, a third-party tool (like Cloudflare) can be configured to serve a cached version of your website while the domain is being reconfigured (for example with `stale-if-error` option). If you choose this solution, be sure to disable caching on routes starting with `/.well-known/acme-challenge` to avoid issues described in the "Technical details and troubleshooting" section.
```